Documentation/Reference.md

```diff
@@ -61,51 +61,117 @@ Where `foo` is an executable and `bar.a` a static library.
 
 ## Package Manifest File Format Reference
 
-Instructions for how to build a package are provided by a manifest file, called `Package.swift`. You can customize this file to declare build targets or dependencies, include or exclude source files, and specify build configurations for the module or individual files.
+Instructions for how to build a package are provided by the `Package.swift` manifest file. `Package.swift` is a Swift file that declaratively configures a Package using types defined in the `PackageDescription` module. It currently has following properties:
 
-Here's an example of a `Package.swift` file:
+### name
+
+Defines the name of the package and is the minimal requirement for a manifest to be valid. When the sources are located directly under `Sources/` directory, there is only one module and the module name is same as the package name.
+
+```swift
+import PackageDescription
+
+let package = Package(
+    name: "FooBar"
+)
+```
+
+### targets
+
+Targets property is used when you have more than one module in your package and want to declare inter-dependency between them.
+
+Note: It is also possible to declare target dependencies between a test and regular module.
 
 ```swift
 import PackageDescription
 
 let package = Package(
     name: "Hello",
-    dependencies: [
-        .Package(url: "ssh://git@example.com/Greeter.git", versions: Version(1,0,0)..<Version(2,0,0)),
+    targets: [
+        Target(name: "TestSupport"),
+        Target(name: "Bar", dependencies: ["Foo"]),
+        Target(name: "BarTests", dependencies: ["Bar", "TestSupport"]),
     ]
 )
 ```
 
-A `Package.swift` file is a Swift file that declaratively configures a Package using types defined in the `PackageDescription` module. This manifest declares a dependency on an external package: `Greeter`.
+Here `Foo`, `Bar` and `TestSupport` are modules present under `Sources/` directory and `BarTests` is a test module under `Tests/` directory.
+`Bar` module can import `Foo` module and `BarTests` module can import `Foo`, `Bar` and `TestSupport` modules.
 
-If your package contains multiple targets that depend on each other you will need to specify their interdependencies. Here is an example:
+### dependencies
+
+This is the list of packages that the current package depends on. You can specify url (or local path) to any valid swift package placed under
```

Comment:
`a URL`, `s/swift/Swift/`

"any valid Swift package placed under version control which is tagged appropriately" is a little unwieldy. I think we should break it up, or just leave out the version control part and document it elsewhere.